metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java

```diff
@@ -540,4 +612,50 @@ public ListUrnsResult listUrns(@Nonnull final String entityName, final int start
     result.setEntities(entityUrns);
     return result;
   }
+
+  private Map<EbeanAspectV2.PrimaryKey, EnvelopedAspect> getEnvelopedAspects(final Set<EbeanAspectV2.PrimaryKey> dbKeys) throws Exception {
+    final Map<EbeanAspectV2.PrimaryKey, EnvelopedAspect> result = new HashMap<>();
+    final Map<EbeanAspectV2.PrimaryKey, EbeanAspectV2> dbEntries = _entityDao.batchGet(dbKeys);
+
+    for (EbeanAspectV2.PrimaryKey currKey : dbKeys) {
+
+      final EbeanAspectV2 currAspectEntry = dbEntries.get(currKey);
+
+      if (currAspectEntry == null) {
+        // No aspect found.
+        continue;
+      }
+
+      // Aspect found. Now turn it into an EnvelopedAspect
+      final com.linkedin.entity.Aspect aspect = RecordUtils.toRecordTemplate(com.linkedin.entity.Aspect.class, currAspectEntry
+          .getMetadata());
+      final EnvelopedAspect envelopedAspect = new EnvelopedAspect();
+      envelopedAspect.setName(currAspectEntry.getKey().getAspect());
+      envelopedAspect.setVersion(currAspectEntry.getKey().getVersion());
+      envelopedAspect.setValue(aspect);
+      envelopedAspect.setCreated(new AuditStamp()
+          .setActor(Urn.createFromString(currAspectEntry.getCreatedBy()))
+          .setTime(currAspectEntry.getCreatedOn().getTime())
+      );
+      result.put(currKey, envelopedAspect);
+    }
+    return result;
+  }
+
+  private EnvelopedAspect getKeyEnvelopedAspect(final Urn urn) throws Exception {
```

Comment:
Almost feel like we should stop returning the Key Aspect in the case in which it's missing.. Injecting it just creates more confusion

Reply:
Yeah this was something that bothered me. I returned it for backward compatibility, but seems very counter-intuitive